cpp/include/cudf/detail/iterator.cuh

```diff
@@ -313,6 +313,40 @@ struct scalar_pair_accessor : public scalar_value_accessor<Element> {
   }
 };
 
+/**
+ * @brief Pair accessor for scalar's representation value and validity.
+ *
+ * @tparam Element The type of element in the scalar.
+ */
+template <typename Element>
+struct scalar_representation_pair_accessor : public scalar_value_accessor<Element> {
+  using super_t    = scalar_value_accessor<Element>;
+  using rep_type   = typename Element::rep;
+  using value_type = thrust::pair<rep_type, bool>;
+
+  scalar_representation_pair_accessor(scalar const& scalar_value)
+    : scalar_value_accessor<Element>(scalar_value)
+  {
+  }
+
+  /**
+   * @brief returns a pair with representative value and validity of the scalar.
+   *
+   * @throw `cudf::logic_error` if this function is called in host.
+   *
+   * @return a pair with representative value and validity of the scalar.
+   */
+  CUDA_HOST_DEVICE_CALLABLE
```

Comment:
Why does this need to be host/device callable if it only throws in host code? Can it just be made device only?

Reply:
> Can it just be made device only?

Yes, I believe so. Let me check. (Sorry, I followed the convention in `scalar_pair_accessor::operator()()`.)

Reply:
Fixed. This works.